### DISCUSSION TITLE
getFolderCount and getFolderTrackCount return 0

### DISCUSSION THREAD

**Discussion by klattimer** [Makuna/DFMiniMp3#119] [Q&A]
I'm trying to arrange my files into folders and it appears the device doesn't appear to be able to read folders. Is there any special kind of file system format required?

**Comment by Makuna** [comment]
[API Reference](https://github.com/Makuna/DFMiniMp3/wiki/API-Reference)

There is no getFolderCount() in the API.  Did you mean getTotalFolderCount()?

There are three types of folders supported by the module; all must be in the root.  There is no nesting of folders supported.  
- one named "mp3" and has specific commands to use it.  Tracks are 4 digit zero padded.  
- one named "advert" and has specific commands to use it. Tracks are 4 digit zero padded.  
- ones named by number which follow two possible formats depending on the module.  Both are zero padded names, one is two digits and the other is three digits.
"001" - older versions, three digit zero padded folder names (experiment to demonstrate if this is what yours needs)
"01" - newer versions, two digit zero padded folder names (experiment to demonstrate if this is what yours needs)
Files within these must be named using a zero padded name with either 3 digits or 4 digits at the front with anything after it.
"001_FunnySound.mp3" - [see playFolderTrack()](https://github.com/Makuna/DFMiniMp3/wiki/API-Reference#void-playfoldertrackuint8_t-folder-uint8_t-track)
"0001_FunnySound.mp3" - [see playFolderTrack16()](https://github.com/Makuna/DFMiniMp3/wiki/API-Reference#void-playfoldertrack16uint8_t-folder-uint16_t-track)

**Reply by klattimer** [reply]
I've re-arranged my folders as mentioned, but I still seem to be having some trouble with making a call to getFolderTrackCount which is always returning 0. I'm not sure but it almost seems like my replies aren't coming back from the device - however I get SD card insert/remove notifications.

**Comment by klattimer** [comment]
Will try this thanks.

**Reply by Makuna** [reply]
What platform are you using?
Turn on [debug feature](https://github.com/Makuna/DFMiniMp3/wiki/Debug-Features) which will give us what the data stream is on the serial monitor.  Then capture and paste the serial output here (use the "code" formatting around it listed in icon bar directly above the edit text are when you reply.
